app/assets/javascripts/template_invocation.js

```diff
@@ -24,6 +24,33 @@ function refresh_search_query(value){
   $('textarea#targeting_search_query').val($('span#bookmark_query_map span#bookmark-' + id).data('query'));
 }
 
+function show_preview_hosts_modal() {
+  var modal_window = $('#previewHostsModal');
+
+  var form = $('form#job_invocation_form');
+  var data = form.serializeArray();
+
+  request = $.ajax({
+    data: data,
+    type: 'POST',
```

Comment:
I have to send the form back to be able to render the hosts list based on the current value of your query